Marlin/dogm_lcd_implementation.h

```diff
@@ -279,26 +279,32 @@ static void lcd_implementation_init() {
 
 static void lcd_implementation_clear() { } // Automatically cleared by Picture Loop
 
+static void _draw_centered_temp(int temp, int x, int y) {
+  int degsize = 6 * (temp >= 100 ? 3 : temp >= 10 ? 2 : 1); // number's pixel width
+  u8g.setPrintPos(x - (18 - degsize) / 2, y); // move left if shorter
+  lcd_print(itostr3(temp));
+  lcd_printPGM(PSTR(LCD_STR_DEGREE " "));
+}
+
```

Comment:
Be careful with calculations here. This is time critical. 3 heaters * ( is + target) * stripes -> 12 for RRDFGD (ST7920) or 48 for the others.